xla/service/gpu/triton_fusion_analysis.cc

```diff
@@ -320,6 +320,16 @@ absl::Status TritonFusionAnalysis::ExecuteForDotFusion(
   return absl::OkStatus();
 }
 
+TritonFusionAnalysis::Scope TritonFusionAnalysis::QueryInstructionScope(
+    const HloInstruction& hlo) const {
+  for (const Scope& scope : {Scope::LHS, Scope::RHS, Scope::OUTPUT}) {
+    if (iter_specs_.at(scope).count(&hlo) > 0) {
+      return scope;
+    }
+  }
+  LOG(FATAL) << "No scope for hlo: " << hlo.ToString();
```

Comment:
LOG(FATAL) seems a bit much, maybe use optional? Do we really want to crash a process there?

Reply:
Maybe we can leave to the util-user to decide whether to crash it. The util will just return nullopt if no scope is found. Done.